PocketKit/Sources/PocketKit/PocketNotificationService.swift

```diff
@@ -0,0 +1,217 @@
+//
+//  PocketNotificationService.swift
+//  
+//
+//  Created by Daniel Brooks on 8/27/22.
+//
+
+import Foundation
+import BrazeKit
+import BrazeUI
+import Sync
+import Analytics
+import SharedPocketKit
+import Combine
+import UIKit
+import UserNotifications
+
+class PocketNotificationService: NSObject {
+    
+    /**
+    The Braze service object
+     */
+    private let braze: Braze
+    
+    /**
+     Instance of our API client for pulling in data when we recieve pushes.
+     */
+    private let source: Source
+    
+    /**
+     Instance of the Pocket Analytics tracker
+     */
+    private let tracker: Tracker
+    
+    /**
+     Instance of the Pocket Session manager for us to subscribe to
+     */
+    private let sessionManager: AppSession
+    
+    /**
+     Subscriptions that this object listens to.
+     */
+    private var subscriptions: Set<AnyCancellable> = []
+
+
+    init(source: Source, tracker: Tracker, sessionManager: AppSession) {
+        self.source = source
+        self.tracker = tracker
+        self.sessionManager = sessionManager
+        
+        // Init Braze with our information.
+        var configuration = Braze.Configuration(
+            apiKey: Keys.shared.brazeAPIKey,
+          endpoint: Keys.shared.brazeAPIEndpoint
+        )
+
+        // Enable logging of general SDK information (e.g. user changes, etc.)
+        configuration.logger.level = .info
+        configuration.push.appGroup = "group.com.ideashower.ReadItLaterProAlp"
+        self.braze = Braze(configuration: configuration)
+        super.init()
+
+        // Set up the in app message ui
+        let inAppMessageUI = BrazeInAppMessageUI()
+        inAppMessageUI.delegate = self
+        self.braze.inAppMessagePresenter = inAppMessageUI
+
+
+        self.sessionManager.$currentSession.sink { session in
+            if let session = session {
+                DispatchQueue.main.async {
+                    // Braze SDK docs say this needs to be called from the main thread.
```

Comment:
We can leverage some conveniences from `Combine` to have this called on the main thread, e.g:

```swift
myPublisher.recieve(on: DispatchQueue.main).sink { val in
    // will be executed on main thread
}
```